packages/core/js-client/src/connection/RelayConnection.ts

```diff
@@ -178,30 +180,39 @@ export class RelayConnection implements IConnection {
 
         await this.lib2p2Peer.handle(
             [PROTOCOL_NAME],
-            async ({ connection, stream }) => {
-                pipe(
-                    stream.source,
-                    // @ts-ignore
-                    decode(),
-                    // @ts-ignore
-                    (source) => map(source, (buf) => toString(buf.subarray())),
-                    async (source) => {
-                        try {
-                            for await (const msg of source) {
-                                try {
-                                    const particle = Particle.fromString(msg);
-                                    log.trace('got particle from stream with id %s and particle id %s', stream.id, particle.id);
+            async ({ connection, stream }) => pipe(
+                stream.source,
+                decode(),
+                (source) => map(source, (buf) => toString(buf.subarray())),
+                async (source) => {
+                    try {
+                        for await (const msg of source) {
+                            try {
+                                const particle = Particle.fromString(msg);
+                                const initPeerId = peerIdFromString(particle.initPeerId);
+                                
+                                if (initPeerId.type !== 'Ed25519') {
```

Comment:
Why? So you are not accepting particles that are generated from rsa or secp256k1? Currently you can easily start nox with such keypairs

Reply:
I wonder if libp2p-js supports that.

Reply:
[yes](https://github.com/libp2p/js-libp2p/blob/91842c93b216a51b65bc79b5248f8e459a646b51/packages/interface/src/peer-id/index.ts#L28)